### PR TITLE
Saga cleanup

### PR DESCRIPTION
**Pull request by RoriCremer**
Tech debt collection!

### CODE REVIEW

File: src/components/DatasetCreateView.jsx [hunk 1]
```diff
@@ -139,7 +139,9 @@ export class DatasetCreateView extends React.PureComponent {
 
   addUser(newEmail) {
     const { dispatch, readers } = this.props;
-    if (!_.includes(readers, newEmail)) {
+    if (!readers) {
+      dispatch(actions.change('dataset.readers', [newEmail]));
+    } else if (!_.includes(readers, newEmail)) {
```
Comment (ddietterich) [inline src/components/DatasetCreateView.jsx:144]: Does this dispatch in the case where readers includes newEmail?

**Review by ddietterich — COMMENTED** [review]
LGTM (which doesn't mean much...)

**Review by ahaessly — APPROVED** [review]
👍

**Review by thathert — APPROVED** [review]
LGTM

File: src/components/DatasetCreateView.jsx (hunk 1 above)
Comment (RoriCremer) [reply src/components/DatasetCreateView.jsx:144]: Where it does not

**Pull request merged**